frontend/public/components/start-guide.tsx

```diff
@@ -10,39 +10,27 @@ import { ProjectModel, RoleModel, StorageClassModel } from '../models';
 
 const WHITELIST = [RoleModel.kind, StorageClassModel.kind];
 
-const seenGuide: string = 'seenGuide';
-
-export class StartGuide extends React.Component<StartGuideProps, StartGuideState> {
-  constructor(props) {
-    super(props);
-
-    // TODO: The dismissable logic is not currently being used, but we'll
-    // probably want a general start guide for OpenShift separate from the no
-    // projects message. Leaving this for now.
-    this.dismiss = this.dismiss.bind(this);
-    let {visible} = props;
-    try {
-      visible = visible || !localStorage.getItem(seenGuide);
-    } catch (ignored) {
-      // ignored
-    }
-    this.state = {
-      visible,
-    };
+export const StartGuide: React.FC<StartGuideProps> = (props) => {
+  let {localStorageKey, startGuide, visible} = props;
+  const seenGuide: string = localStorageKey ? localStorageKey : 'seenGuide';
```

Comment:
I don't think it's a good idea to default `localStorageKey`. That really needs to be unique for each guide. I suggest making it a single prop like `dismissKey` and make it dismissible when that's provided.

Reply:
I got rid of the default key and made it a required prop instead. I think we need two props - dismissible and dismissKey (or localStorageKey) since all start guides need keys, but only some have dismiss buttons.

Reply:
Why do start guides that can't be dismissed need a key? It seems to only be used to track the dismiss state in local storage.

Reply:
I assumed it needed it since it was there. Clearly it doesn't. Updated in new commit.